File: frontend/src/app/controller/service/admin/adresse/RueAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {RueDto} from 'src/app/controller/model/adresse/Rue.model';
import {RueCriteria} from 'src/app/controller/criteria/adresse/RueCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class RueAdminService extends AbstractService<RueDto, RueCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/rue/');
    }

    public constrcutDto(): RueDto {
        return new RueDto();
    }

    public constrcutCriteria(): RueCriteria {
        return new RueCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/univ/service/facade/admin/adresse/RueAdminService.java
package ma.zs.univ.service.facade.admin.adresse;

import java.util.List;
import ma.zs.univ.bean.core.adresse.Rue;
import ma.zs.univ.dao.criteria.core.adresse.RueCriteria;
import ma.zs.univ.zynerator.service.IService;



public interface RueAdminService extends  IService<Rue,RueCriteria>  {

    List<Rue> findByQuartierId(Long id);
    int deleteByQuartierId(Long id);
    long countByQuartierCode(String code);




}
